src/arr/compiler/resolve-scope.arr

```diff
@@ -231,8 +387,10 @@ fun desugar-scope-block(stmts :: List<A.Expr>, binding-group :: BindingGroup) ->
           add-letrec-bind(binding-group, A.s-letrec-bind(
               l,
               A.s-bind(l, false, A.s-name(l, name), A.a-blank),
-              A.s-lam(l, name, params, args, ann, doc, body, _check-loc, _check, blocky)
-            ), rest-stmts)
+              # NOTE(Ben): deliberately keeping this as an s-fun;
+              # it'll get turned into an s-lam in weave-contracts
```

Comment:
This comment says _what_ is happening, but not _why_. I think the reason is that we want to give errors for contract-ing anything that isn't an `s-fun`, right?